src/internals.cc

```diff
@@ -22,10 +22,37 @@
 
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
+#include <gul14/join_split.h>
+#include <gul14/SmallVector.h>
+
 #include "internals.h"
 
 namespace task {
 
 const gul14::string_view abort_marker{ u8"\U0001F6D1ABORT\U0001F6D1" };
 
+ErrorCause remove_abort_markers_from_error_message(std::string& msg)
+{
+    const auto tokens = gul14::split<gul14::SmallVector<gul14::string_view, 3>>(
+        msg, abort_marker);
+
+    if (tokens.size() >= 3) // The real error message is between the first 2 abort markers
+        msg.assign(tokens[1].begin(), tokens[1].end());
+    else
+        msg = gul14::join(tokens, "");
+
+    if (tokens.size() > 1) // There was at least one abort marker
+    {
+        if (msg.empty())
+        {
+            msg = "Script called terminate_sequence()";
+            return ErrorCause::terminated_by_script;
+        }
+
+        return ErrorCause::aborted;
+    }
```

Comment:
Rewrote this to take a `string_view` as an input parameter, and then we automatically have to use some logic like this. Unfortunately not quite as nice because of several `string_view`->`string` conversions...